Dockerfile

```diff
@@ -0,0 +1,19 @@
+FROM python:3.8.4-buster as base
+ENV POETRY_HOME=/poetry
+ENV PATH=${POETRY_HOME}/bin:${PATH}
+RUN curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python
+EXPOSE 5000
+WORKDIR /todo-apprenticeship
+COPY . /todo-apprenticeship/
```

Comment:
There's nothing wrong with this broad COPY approach, but obviously it's more at risk of someone having additional files that we might want to Dockerignore  than a more targeted set up; worth weighing up whether using a whitelist at the root level combined with a broader copy at the code level might be nicer;
```Dockerfile
COPY pyproject.toml poetry.lock /todo-apprenticeship/
COPY ./todo_app /todo-apprenticeship/todo_app/
```

This also has a slight advantage that it naturally separates your dependencies from your code - allowing the Docker cache to be a little more effective _if_ it's true that your code is changing more often than your dependencies (e.g. if we introduced the `poetry install` between the two commands, we could cache the installation step better)